K2Bridge/KustoDAL/HitsMapper.cs

```diff
@@ -59,11 +59,18 @@ private static IEnumerable<Hit> MapAndAnalyzeRows(DataRowCollection kustoRespons
         /// <param name="query">QueryData containing query information.</param>
         /// <param name="highlighter">Lucene Highligher.</param>
         /// <returns>Hit model.</returns>
+        /// <remarks>If there is no "_id" field in Kusto data, it will be a random integer.</remarks>
         private static Hit ReadHit(DataRow row, QueryData query, LuceneHighlighter highlighter)
         {
             Ensure.IsNotNull(row, nameof(row));
 
-            var hit = HitsFactory.Create(Random.Next().ToString(), query.IndexName);
+            object id = null;
+            if (row.Table.Columns.Contains("_id"))
+            {
+                id = row["_id"];
+            }
```

Comment:
Hi @boazsha.
We've added unit tests, but here's the catch:
In `MapsRowsToHits_WithoutIdInSample_ReturnsSomeRandomNonEmptyArray` we test that the Id is not empty, so we imply random _id was generated in line 73(`Random.Next().ToString()`). We thought that it would be nice to be able to mock Random here and check the exact value in test. This would require many changes to HitsMapper, so we decided not to do this now. What do you think?

Reply:
Thanks @spacentropy!
I think this is great, but we should definitely consider making our classes mockable.